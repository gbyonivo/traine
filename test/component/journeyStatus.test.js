import React from 'react';
import { create } from 'react-test-renderer';
import JourneyStatus from '../../src/components/journeyStatus';

const realTimeUpdatesInfo = {
  realTimeServiceInfo: {
    realTimeUpdatesInfo: {
      realTime: '2018-07-19T19:06:00+01:00',
      realTimePlatform: '3',
      realTimeFlag: 'Estimate'
    }
  }
};

const scheduleInfo = {
  scheduledTime: '2018-07-19T19:05:00+01:00',
};

const onTimeScheduleInfo = {
  scheduledTime: '2018-07-19T19:06:00+01:00',
};

const delayedData = {
  realTimeUpdatesInfo: {
    ...realTimeUpdatesInfo,
    ...{
      delayReason: {
        code: '186',
        reasonText: 'This train has been delayed by waiting for a train crew member'
      }
    }
  }
};

describe('JourneyStatus', () => {
  it('should render correctly and display delay message', () => {
    const actual = create(<JourneyStatus realTimeUpdatesInfo={delayedData.realTimeUpdatesInfo} scheduledInfo={{}} />);
    expect(actual).toMatchSnapshot();
  });
  it('should render correctly and display exp message', () => {
    const actual = create(<JourneyStatus realTimeUpdatesInfo={delayedData.realTimeUpdatesInfo} scheduledInfo={scheduleInfo} />);
    expect(actual).toMatchSnapshot();
  });
  it('should render correctly and display ontime', () => {
    const actual = create(<JourneyStatus realTimeUpdatesInfo={delayedData.realTimeUpdatesInfo} scheduledInfo={onTimeScheduleInfo} />);
    expect(actual).toMatchSnapshot();
  });
  it('should render correctly and display on time', () => {
    const actual = create(<JourneyStatus realTimeUpdatesInfo={{}} scheduledInfo={onTimeScheduleInfo} />);
    expect(actual).toMatchSnapshot();
  });
});