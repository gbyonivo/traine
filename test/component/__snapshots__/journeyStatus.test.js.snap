// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JourneyStatus should render correctly and display delay message 1`] = `
<div
  className="delayed undefined journeyStatus"
>
  Delayed
</div>
`;

exports[`JourneyStatus should render correctly and display exp message 1`] = `
<div
  className="delayed undefined journeyStatus"
>
  Delayed
</div>
`;

exports[`JourneyStatus should render correctly and display on time 1`] = `
<div
  className="onTime undefined journeyStatus"
>
  On Time
</div>
`;

exports[`JourneyStatus should render correctly and display ontime 1`] = `
<div
  className="delayed undefined journeyStatus"
>
  Delayed
</div>
`;
